src/handlers/spell.handler.ts

```diff
@@ -0,0 +1,35 @@
+import { Socket } from "socket.io";
+import { logger } from "../logger";
+import { UNAUTHORIZED } from "../models/gameerror";
+import { game } from "../server";
+import { MessageHandler } from "./handler";
+
+export class SpellHandler extends MessageHandler {
+  constructor(socket: Socket) {
+    super(socket);
+  }
+
+  handleMessage(message: string): void {
+    if (game.state.status !== "PLAYING") {
+      this.socket.emit("error", JSON.stringify(UNAUTHORIZED));
+      return;
+    }
+
+    const {
+      playerId,
+      id,
+      name,
+      cooldown,
+    }: { playerId: string; id: number; name: string; cooldown: number } =
+      JSON.parse(message);
+
+    const player = game.players[playerId];
```

Comment:
We might need some checks to make sure the playerId exists and also that the spells exists